core/domain/classifier_services.py

```diff
@@ -359,62 +416,56 @@ def delete_classifier_training_job(job_id):
         classifier_training_job_model.delete()
 
 
-def get_classifier_from_exploration_attributes(exp_id, exp_version,
-                                               state_name):
-    """Gets the classifier object from the exploration attributes.
+def get_classifier_training_jobs(exp_id, exp_version, state_names):
+    """Gets the classifier training job object from the exploration attributes.
 
     Args:
         exp_id: str. ID of the exploration.
         exp_version: int. The exploration version.
-        state_name: str. The name of the state to which the classifier
-            belongs.
+        state_names: list(str). The state names for which we retrieve the job.
 
     Returns:
-        ClassifierData. Domain object for the Classifier model.
+        list(ClassifierTrainingJob). Domain objects for the Classifier training
+            job model.
 
     Raises:
-        Exception: Entity for class ClassifierExplorationMapping with id not
+        Exception: Entity for class TrainingJobExplorationMapping with id not
```

Comment:
@seanlip This one here is tricky. So, first we rettrieve all mapping_models. In the case where job doesn't exist, mapping model doesn't exist either. So, the get_models() method returns a list where if an element is None, we don't have a mapping. So, to maintain indexes, we need to append something to the list of job_ids so that get_multi(job_ids) is consistent. get_multi() cannot take None as any element in the list. So, the only way I could get it to work was to explicitly append a false job_id string whenever the mapping model didn't exist. If we could do that, then indexes are maintained and the list of jobs that get returned at the very end works the way we want it to (with None in indexes where job doesn't exist). So, is this fine?